lib/SILGen/SILGenExpr.cpp

```diff
@@ -2293,6 +2293,20 @@ RValue RValueEmitter::visitRebindSelfInConstructorExpr(
   return SGF.emitEmptyTupleRValue(E, C);
 }
 
+static bool isNullableTypeInC(Type ty) {
+  if (ty->hasReferenceSemantics())
+    return true;
+  
+  auto &C = ty->getASTContext();
+  auto nom = ty->getAnyNominal();
+  if (!nom)
+    return false;
+
+  return nom == C.getUnsafePointerDecl()
+    || nom == C.getUnsafeMutablePointerDecl()
+    || nom == C.getAutoreleasingUnsafeMutablePointerDecl();
+}
```

Comment:
Sorry, there are several more pointer types: OpaquePointer, Selector, and…well, I guess we don't have to care about NSZone.

I'm not happy about this list being hardcoded in more and more places, though. I suppose it's not enough to check the Clang type because that would include things like va_list, which might coincidentally be pointers.